app/services/fees/create_instant_service.rb

```diff
@@ -10,11 +10,105 @@ def initialize(charge:, event:)
     end
 
     def call
+      fees = []
+
+      ActiveRecord::Base.transaction do
+        if charge.group_properties.blank?
+          fees << create_fee(properties: charge.properties)
+        else
+          charge.group_properties.each do |group_properties|
+            group = billable_metric.selectable_groups.find_by(id: group_properties.group_id)
+            next unless event_linked_to?(group:)
+
+            fees << create_fee(properties: group_properties.values, group:)
+          end
+        end
+      end
+
+      result.fees = fees.compact
+      # TODO(instant_charges): deliver webhooks for fees
+
       result
+    rescue BaseService::FailedResult => e
+      result.fail_with_error!(e)
+    rescue ActiveRecord::RecordInvalid => e
+      result.record_validation_failure!(record: e.record)
     end
 
     private
 
     attr_reader :charge, :event
+
+    delegate :billable_metric, to: :charge
+    delegate :subscription, :customer, to: :event
+
+    def create_fee(properties:, group: nil)
+      aggregation_result = aggregate(properties:, group:)
+      result = apply_charge_model(aggregation_result:, properties:)
+
+      fee = Fee.new(
```

Comment:
`event.subscription.fees.new`?

Reply:
@rsempe I prefer not to apply this one :)

I found it easier to read, and easier to find.